src/report/report-generate.controller.js

```diff
@@ -0,0 +1,286 @@
+                ),
+                '_blank'
+            );
+        }
+
+        /**
+         * @ngdoc method
+         * @methodOf report.controller:ReportGenerateController
+         * @name watchDependency
+         *
+         * @description
+         * Sets up a watch on report parameter selection,
+         * to update dependent parameters options based on it's value.
+         *
+         * @param   {Object}    param             the report parameter that needs to watch for
+         *                                        dependency.
+         * @param   {Object}    dep               the dependency object to set up watch for.
+         */
+        function watchDependency(param, dep) {
+            var watchProperty = 'vm.selectedParamsOptions.' + dep.dependency;
+            $scope.$watch(watchProperty, function(newVal) {
+                vm.selectedParamsDependencies[dep.dependency] = newVal;
+                if (newVal) {
+                    reportFactory.getReportParamOptions(param, vm.selectedParamsDependencies)
+                        .then(function(items) {
+                            vm.paramsOptions[param.name] = items;
+                        });
+                }
+            });
+        }
+
+        /**
+         * @ngdoc method
+         * @methodOf report.controller:ReportGenerateController
+         * @name $onInit
+         *
+         * @description
+         * Initialization method of the ReportGenerateController.
+         */
+        function onInit() {
+            angular.forEach(report.templateParameters, function(param) {
+                angular.forEach(param.dependencies, function(dependency) {
+                    watchDependency(param, dependency);
+                });
+            });
+            // SELVSUP-6: Create Stock on Hand Report on reports page
+            watchFacilitySelection();
+            vm.filterAvailableParameters = filterAvailableParameters;
+            vm.mapParameters = mapParameters;
+            // SELVSUP-6: Ends here
+        }
+
+        // SELVSUP-6: Create Stock on Hand Report on reports page
+        function changeCommasToSemicolons(text) {
+            if (text === null) {
+                return text;
+            }
+            var reducedText = text.reduce(function(acc, curr) {
+                return acc + curr.replace(',', ';;') + ';';
+            }, '');
+            return reducedText.slice(0, reducedText.length - 1);
+        }
+
+        /**
+         * @ngdoc method
+         * @methodOf report.controller:ReportGenerateController
+         * @name mapParameters
+         *
+         * @description
+         * Method changig the types of parameters in order to create an report (which requires concrete types)
+         * @param  {Array} parameter the parameter used when filtering tags
+         * @return {String} string 
+         */
+        function mapParameters(parameters) {
+            var mappedParameters = Object.assign({}, parameters);
```

Comment:
Is `parameters` always an object with prmitive types (like strings)?
If that's the case no changes are needed, otherwise you could consider using `angular.copy` instead to create a deep copy.

Reply:
Params should be always strings